python/ray/util/spark/cluster_init.py

```diff
@@ -1172,6 +1199,181 @@ def setup_ray_cluster(
     return cluster.address
 
 
+def _start_ray_worker_nodes(
+    *,
+    spark,
+    spark_job_group_id,
+    spark_job_group_desc,
+    num_worker_nodes,
+    using_stage_scheduling,
+    ray_head_ip,
+    ray_head_port,
+    ray_temp_dir,
+    num_cpus_per_node,
+    num_gpus_per_node,
+    heap_memory_per_node,
+    object_store_memory_per_node,
```

Comment:
can they be put inside `worker_node_options` as well?

Reply:
I suggest we make these as separate arguements, these are important ray node configs which might require some more preprocessing logic, e.g., for `--num-gpus` options, we need to generate `CUDA_VISIBLE_DEVICES` environment variable for ray worker node.